File: web/src/api/kubernetes/deployment.js
import service from '@/utils/request'

// 获取deployment list
export const getDeploymentList = (data) => {
  return service({
    url: '/deployment/getDeploymentList',
    method: 'post',
    data
  })
}

// 获取deployment in raw
export const getDeploymentRaw = (data) => {
  return service({
    url: '/deployment/getDeploymentRaw',
    method: 'post',
    data
  })
}

// 获取deployment in detail
export const getDeploymentDetail = (data) => {
  return service({
    url: '/deployment/getDeploymentDetail',
    method: 'post',
    data
  })
}

// 获取deployment关联的replicaset
export const getNewReplicaSet = (data) => {
  return service({
    url: '/deployment/getNewReplicaSet',
    method: 'post',
    data
  })
}

// 删除deployment
export const deleteDeployment = (data) => {
  return service({
    url: '/deployment/deleteDeployment',
    method: 'post',
    data
  })
}

File: server/router/kubernetes/deployment.go
package kubernetes

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/pddzl/kubeops/server/api/v1"
	"github.com/pddzl/kubeops/server/middleware"
)

type DeploymentRouter struct{}

func (d *DeploymentRouter) InitDeploymentRouter(Router *gin.RouterGroup) {
	deploymentRouter := Router.Group("deployment").Use(middleware.OperationRecord())
	deploymentRouterWithoutRecord := Router.Group("deployment")
	deploymentApi := v1.ApiGroupApp.KubernetesApiGroup.DeploymentApi
	{
		deploymentRouter.POST("getDeploymentDetail", deploymentApi.GetDeploymentDetail) // 获取deployment详情
		deploymentRouter.POST("getDeploymentRaw", deploymentApi.GetDeploymentRaw)       // 获取deployment in 编排
		deploymentRouter.POST("getNewReplicaSet", deploymentApi.GetNewReplicaSet)       // 获取deployment关联的replicaSet
		deploymentRouter.POST("deleteDeployment", deploymentApi.DeleteDeployment)       // 删除deployment
	}
	{
		deploymentRouterWithoutRecord.POST("getDeploymentList", deploymentApi.GetDeploymentList) // 获取所有deployment
	}
}
